addons/osio-addon/src/main/java/io/fabric8/launcher/osio/jenkins/JenkinsGitCredentials.java

```diff
@@ -0,0 +1,101 @@
+package io.fabric8.launcher.osio.jenkins;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.RequestScoped;
+import javax.inject.Inject;
+import javax.json.Json;
+import javax.json.JsonObjectBuilder;
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.core.HttpHeaders;
+
+import io.fabric8.launcher.base.http.ExternalRequest;
+import io.fabric8.launcher.base.identity.Identity;
+import io.fabric8.launcher.base.identity.IdentityVisitor;
+import io.fabric8.launcher.base.identity.TokenIdentity;
+import io.fabric8.launcher.core.spi.IdentityProvider;
+import io.fabric8.launcher.osio.EnvironmentVariables;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+
+import static io.fabric8.utils.URLUtils.pathJoin;
+
+@RequestScoped
+public class JenkinsGitCredentials {
+    public JenkinsGitCredentials() {
+    }
+
+    JenkinsGitCredentials(String authorization, String gitToken) {
+        this.authorization = authorization;
+        this.gitToken = gitToken;
+    }
+
+    static final String GET_URL = pathJoin(EnvironmentVariables.getJenkinsUrl(), "/credentials/store/system/domain/_/credentials/cd-github/");
+
+    static final String CREATE_URL = pathJoin(EnvironmentVariables.getJenkinsUrl(), "/credentials/store/system/domain/_/createCredentials");
+
+    private String authorization;
+
+    private String gitToken;
+
+    @Inject
+    private HttpServletRequest request;
+
+    @Inject
+    private IdentityProvider identityProvider;
+
+    @PostConstruct
+    public void init() {
+        authorization = request.getHeader(HttpHeaders.AUTHORIZATION);
+        Identity identity = identityProvider.getIdentity(IdentityProvider.ServiceType.GITHUB, authorization)
+                .orElseThrow(() -> new IllegalStateException("Invalid GITHUB token"));
+        identity.accept(new IdentityVisitor() {
+
+            @Override
+            public void visit(TokenIdentity token) {
+                JenkinsGitCredentials.this.gitToken = token.getToken();
+            }
+        });
+    }
+
+    public void ensureCredentials(String gitUserName) {
+        if (!credentialsExist()) {
+            createCredentials(gitUserName);
+        }
+    }
+
+    private boolean credentialsExist() {
+        Request getRequest = new Request.Builder()
+                .url(GET_URL)
+                .header(HttpHeaders.AUTHORIZATION, authorization)
+                .build();
+
+        return ExternalRequest.execute(getRequest, okhttp3.Response::isSuccessful);
+    }
+
+    private void createCredentials(String gitUserName) {
+
+        JsonObjectBuilder credentials = Json.createObjectBuilder();
+        credentials.add("", 0);
```

Comment:
Is this necessary?

Reply:
I don't know, I just copy paste

Reply:
Can you test if that is needed? We really want to avoid inheriting technical debt